pkgs/development/compilers/emscripten/emscripten.nix

```diff
@@ -0,0 +1,136 @@
+    })
+    # https://github.com/emscripten-core/emscripten/pull/18220
+    (fetchpatch {
+      url = "https://github.com/emscripten-core/emscripten/commit/852982318f9fb692ba1dd1173f62e1eb21ae61ca.patch";
+      sha256 = "sha256-hmIOtpRx3PD3sDAahUcreSydydqcdSqArYvyLGgUgd8=";
+    })
+  ];
+
+  buildPhase = ''
+    runHook preBuild
+
+    patchShebangs .
+
+    # fixes cmake support
+    sed -i -e "s/print \('emcc (Emscript.*\)/sys.stderr.write(\1); sys.stderr.flush()/g" emcc.py
+
+    # disables cache in user home, use installation directory instead
+    sed -i '/^def/!s/root_is_writable()/True/' tools/config.py
+    sed -i "/^def check_sanity/a\\  return" tools/shared.py
+
+    # required for wasm2c
+    ln -s ${nodeModules} node_modules
+
+    echo "EMSCRIPTEN_ROOT = '$out/share/emscripten'" > .emscripten
+    echo "LLVM_ROOT = '${llvmEnv}/bin'" >> .emscripten
+    echo "NODE_JS = '${nodejs}/bin/node'" >> .emscripten
+    echo "JS_ENGINES = [NODE_JS]" >> .emscripten
+    echo "CLOSURE_COMPILER = ['$closurecompiler/bin/closure-compiler']" >> .emscripten
+    echo "JAVA = '$jre/bin/java'" >> .emscripten
+    # to make the test(s) below work
+    # echo "SPIDERMONKEY_ENGINE = []" >> .emscripten
+    echo "BINARYEN_ROOT = '${binaryen}'" >> .emscripten
+
+    # make emconfigure/emcmake use the correct (wrapped) binaries
+    sed -i "s|^EMCC =.*|EMCC='$out/bin/emcc'|" tools/shared.py
+    sed -i "s|^EMXX =.*|EMXX='$out/bin/em++'|" tools/shared.py
+    sed -i "s|^EMAR =.*|EMAR='$out/bin/emar'|" tools/shared.py
+    sed -i "s|^EMRANLIB =.*|EMRANLIB='$out/bin/emranlib'|" tools/shared.py
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    appdir=$out/share/emscripten
+    mkdir -p $appdir
+    cp -r . $appdir
+    chmod -R +w $appdir
+
+    mkdir -p $out/bin
+    for b in em++ em-config emar embuilder.py emcc emcmake emconfigure emmake emranlib emrun emscons emsize; do
+      makeWrapper $appdir/$b $out/bin/$b \
+        --set NODE_PATH ${nodeModules} \
+        --set EM_EXCLUSIVE_CACHE_ACCESS 1 \
+        --set PYTHON ${python3}/bin/python \
+        --set PYTHONPATH '${python3Packages.makePythonPath [ python3Packages.requests ]}'
+    done
+
+    export EM_CACHE=$out/share/emscripten/cache
+    # Minimal cache dependencies needed to run test_hello_world
+    $out/bin/embuilder.py build \
+      libGL libal libhtml5 libstubs-debug libnoexit \
+      libc-debug libdlmalloc libcompiler_rt libsockets \
+      struct_info libwasmfs-debug libc++-noexcept \
+      libc++abi-debug-noexcept
+
+    export EM_FROZEN_CACHE=1
+
+    export PYTHON=${python3}/bin/python
+    export NODE_PATH=${nodeModules}
+    pushd $appdir
+    python test/runner.py test_hello_world
+    popd
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/emscripten-core/emscripten";
+    description = "An LLVM-to-JavaScript Compiler";
+    platforms = platforms.all;
```

Comment:
Not sure if unix would be better here

Reply:
This matches `platforms.all` in the existing `emscripten` package.